inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/thirdparty/sort/util/SinkInfoUtils.java

```diff
@@ -76,13 +80,53 @@ public static SinkInfo createSinkInfo(InlongStreamInfo inlongStreamInfo, SourceR
         String sinkType = sinkResponse.getSinkType();
         if (SinkType.forType(sinkType) == SinkType.HIVE) {
             return createHiveSinkInfo(sourceResponse, (HiveSinkResponse) sinkResponse);
-        } else if (SinkType.forType(sinkType) == SinkType.KAFKA) {
+        }
+        if (SinkType.forType(sinkType) == SinkType.KAFKA) {
             return createKafkaSinkInfo(inlongStreamInfo, sourceResponse, (KafkaSinkResponse) sinkResponse);
-        } else {
-            //todo clickhouse and iceberg is wait to support
+        }
+        if (SinkType.forType(sinkType) == SinkType.CLICKHOUSE) {
+            return createClickhouseSinkInfo(inlongStreamInfo, sourceResponse, (ClickHouseSinkResponse) sinkResponse);
+        }
+        // todo iceberg is wait to support
+        throw new RuntimeException(
+                String.format("SinkType:{} not support in CreateSortConfigListener", sinkType));
+    }
+
+    private static ClickHouseSinkInfo createClickhouseSinkInfo(InlongStreamInfo inlongStreamInfo,
```

Comment:
Are you sure you need a `private static` method?

Reply:
> Are you sure you need a `private static` method?

only for `createSinkInfo` to use